superduperdb/components/component.py

```diff
@@ -16,16 +17,18 @@
     from superduperdb.components.dataset import Dataset
 
 
+@dc.dataclass
 class Component(Serializable):
     """
-    Base component which model, listeners, learning tasks etc. inherit from.
-    """
+    :param identifier: A unique identifier for the component"""
 
     type_id: t.ClassVar[str]
+    identifier: str
 
-    if t.TYPE_CHECKING:
-        identifier: t.Optional[str]
-        version: t.Optional[int]
+    def __post_init__(self) -> None:
+        # set version in `__post_init__` so that is
+        # cannot be set in `__init__` and is always set
+        self.version: t.Optional[int] = None
```

Comment:
Add `self.type_id = self.__class__.__name__` in the post_init

Reply:
I think we need to discuss this -- can be done later. (The code is old, just moved around a bit, creating this diff.)